src/shogun/util/factory.h

```diff
@@ -315,12 +318,12 @@ namespace shogun
 		return result;
 	}
 
-	std::shared_ptr<File> csv_file(std::string fname, char rw = 'r')
+	std::shared_ptr<CSVFile> csv_file(std::string fname, char rw = 'r')
 	{
 		return std::make_shared<CSVFile>(fname.c_str(), rw);
 	}
 
-	std::shared_ptr<File> libsvm_file(std::string fname, char rw = 'r')
+	std::shared_ptr<LibSVMFile> libsvm_file(std::string fname, char rw = 'r')
```

Comment:
Is there a reason why you changed the type here?

Reply:
``` cpp
template <typename TypeName, typename... Args>
std::shared_ptr<TypeName> create(Args&&... args)
```
The **TypeName** of return value should be the same as the **TypeName** of create<TypeName>.